server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSPerformSPOptionsCheckUtility.java

```diff
@@ -0,0 +1,282 @@
+        checkHasMandatoryOptions(ctx.NETNAME(), ctx, "NETNAME");
+    }
+    private void checkJvmServer(CICSParser.Cics_perform_jvmserverContext ctx) {
+        checkHasMandatoryOptions(ctx.JVMSERVER(), ctx, "JVMSERVER");
+        checkHasExactlyOneOption("JVMTYPE or JVM or LIBERTY or OSGI", ctx, ctx.JVMTYPE(), ctx.JVM(), ctx.LIBERTY(), ctx.OSGI());
+        if (!ctx.JVM().isEmpty()) {
+            checkHasExactlyOneOption("JVMACTION or DUMP or GATHER or STACKTRACE", ctx, ctx.JVMACTION(), ctx.DUMP(), ctx.GATHER(), ctx.STACKTRACE());
+            if (!ctx.DUMP().isEmpty()) {
+                checkHasExactlyOneOption("DUMPTYPE or ALL or JAVACORE or HEAP or SNAPTRACE", ctx, ctx.DUMPTYPE(), ctx.ALL(), ctx.JAVACORE(), ctx.HEAP(), ctx.SNAPTRACE());
+            } else if (!ctx.GATHER().isEmpty()) {
+                checkHasExactlyOneOption("GATHERTYPE or DIAGNOSTICS", ctx, ctx.GATHERTYPE(), ctx.DIAGNOSTICS());
+            } else if (!ctx.STACKTRACE().isEmpty() || !ctx.TASKID().isEmpty()) {
+                checkHasMandatoryOptions(ctx.STACKTRACE(), ctx, "STACKTRACE");
+                checkHasMandatoryOptions(ctx.TASKID(), ctx, "TASKID");
+            }
+        } else if (!ctx.LIBERTY().isEmpty()) {
+            checkHasExactlyOneOption("LIBRTYACTION or REFRESH or SERVERDUMP", ctx, ctx.LIBRTYACTION(), ctx.REFRESH(), ctx.SERVERDUMP());
+            if (!ctx.REFRESH().isEmpty()) {
+                checkHasExactlyOneOption("RESOURCETYPE or APPLICATION or CONFIG", ctx, ctx.RESOURCETYPE(), ctx.APPLICATION(), ctx.CONFIG());
+                if (!ctx.APPLICATION().isEmpty() || !ctx.APPID().isEmpty()) {
+                    checkHasMandatoryOptions(ctx.APPLICATION(), ctx, "APPLICATION");
+                    checkHasMandatoryOptions(ctx.APPID(), ctx, "APPID");
+                }
+            }
+            } else if (!ctx.OSGI().isEmpty()) {
+                checkHasExactlyOneOption("OSGIACTION or REFRESHPKGS", ctx, ctx.OSGIACTION(), ctx.REFRESHPKGS());
+            }
+    }
+    private void checkPipeline(CICSParser.Cics_perform_pipelineContext ctx) {
+        checkHasMandatoryOptions(ctx.PIPELINE(), ctx, "PIPELINE");
+        checkHasMutuallyExclusiveOptions("ACTION or SCAN", ctx.ACTION(), ctx.SCAN());
+    }
+    private void checkSecdiscovery(CICSParser.Cics_perform_secdiscoveryContext ctx) {
+        checkHasMandatoryOptions(ctx.SECDISCOVERY(), ctx, "PIPELINE");
+        checkHasExactlyOneOption("ACTION or WRITE", ctx, ctx.ACTION(), ctx.WRITE());
+    }
+    private void checkSecurity(CICSParser.Cics_perform_securityContext ctx) {
+        checkHasMandatoryOptions(ctx.SECURITY(), ctx, "SECURITY");
+        checkHasMandatoryOptions(ctx.REBUILD(), ctx, "REBUILD");
+    }
+    private void checkShutdown(CICSParser.Cics_perform_shutdownContext ctx) {
+        checkHasMandatoryOptions(ctx.SHUTDOWN(), ctx, "SHUTDOWN");
+        checkHasMutuallyExclusiveOptions("SDTRAN or NOSDTRAN", ctx.SDTRAN(), ctx.NOSDTRAN());
+        if (!ctx.SHUTDOWN().isEmpty()) {
```

Comment:
SHUTDOWN is required. The check can be assumed true for the validation purposes.